### DISCUSSION TITLE
Conversion warning - Wifi Video

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#84] [Course Qeustions] [answered]
Very small issue but I get a warning in this line in wifi_connect_sta(...) routine

esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);

The warning states "implicit conversion from enumeration type 'esp_interface_t' to different enumeration type 'wifi_interface_t'
for "ESP_IF_WIFI_STA" (Full function code below)...

esp_err_t wifi_connect_sta(const char *ssid,const char *password,int timeout)
{
	wifi_events = xEventGroupCreate();
	
	esp_netif = esp_netif_create_default_wifi_sta();
	
	wifi_config_t wifi_config;
	memset(&wifi_config, 0, sizeof(wifi_config_t));
	strncpy((char*)wifi_config.sta.ssid, ssid, sizeof(wifi_config.sta.ssid) - 1);
	strncpy((char*)wifi_config.sta.password, password, sizeof(wifi_config.sta.password) - 1);
	
	esp_wifi_set_mode(WIFI_MODE_STA);
	esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
	esp_wifi_start();
	
	EventBits_t result = xEventGroupWaitBits(wifi_events, CONNECTED_GOT_IP / DISCONNECTED, true, false, pdMS_TO_TICKS(timeout));
	if(result == CONNECTED_GOT_IP)
	{ 
		return ESP_OK;
	}
	return ESP_FAIL;
}

**Comment by Mair** [comment]
strange. I don't get that ¯\\_(ツ)_/¯

**Comment by orbitcoms** [comment, marked as answer]
Yeah, and it is only on this one line

esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
